logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/GetOrRegisterClientUseCase.kt

```diff
@@ -0,0 +1,42 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.cryptography.ProteusClient
+import kotlinx.coroutines.flow.firstOrNull
+
+interface GetOrRegisterClientUseCase {
+    suspend operator fun invoke(
+        registerClientParam: RegisterClientUseCase.RegisterClientParam
+    ): RegisterClientResult
+}
+
+class GetOrRegisterClientUseCaseImpl(
+    private val observeCurrentClientId: ObserveCurrentClientIdUseCase,
+    private val selfClients: SelfClientsUseCase,
+    private val registerClient: RegisterClientUseCase,
+    private val clearClientData: ClearClientDataUseCase,
+    private val proteusClient: ProteusClient
+) : GetOrRegisterClientUseCase {
+
+    override suspend fun invoke(registerClientParam: RegisterClientUseCase.RegisterClientParam): RegisterClientResult {
+        val result: RegisterClientResult? = observeCurrentClientId().firstOrNull()?.let { currentClientId ->
```

Comment:
can be replaced with a function that does not return a flow and remove the firstOrNull()

Reply:
Ok, changed 👌